### ISSUE TITLE
Error with variable interpolation and compiled container

### ISSUE THREAD

**Issue by n00dl3** [PHP-DI/PHP-DI#770]
Hey, thanks for this nice piece of software !

I created a factory for my Doctrine Mongo connection and I realized that some string interpolation are not working when compiled .

this is the original factory :

```php
    Client::class => function (ContainerInterface $c) {
        $mongoPort = $c->get('settings.mongo.port');
        $mongoHost = $c->get('settings.mongo.host');
        $mongoDB = $c->get('settings.mongo.dbName');
        $mongoUser = $c->get('settings.mongo.user');
        $mongoPassword = $c->get('settings.mongo.password');
        return new Client("mongodb://${mongoHost}:${mongoPort}/${mongoDB}", [
            'username' => $mongoUser,
            'password' => $mongoPassword,
        ], ['typeMap' => DocumentManager::CLIENT_TYPEMAP]);
    },
```

But once compiled it becomes something like this :

```php
protected function get67()
    {
        return $this->resolveFactory(static function (\Psr\Container\ContainerInterface $c) {
        $mongoPort = $c->get('settings.mongo.port');
        $mongoHost = $c->get('settings.mongo.host');
        $mongoDB = $c->get('settings.mongo.dbName');
        $mongoUser = $c->get('settings.mongo.user');
        $mongoPassword = $c->get('settings.mongo.password');
        return new \MongoDB\Client("mongodb://{mongoHost}:{mongoPort}/{mongoDB}", [
            'username' => $mongoUser,
            'password' => $mongoPassword,
        ], ['typeMap' => \Doctrine\ODM\MongoDB\DocumentManager::CLIENT_TYPEMAP]);
    }, 'MongoDB\\Client');
    }
```
As you can see, the string interpolation is (badly) rewritten, so I tried other interpolation syntaxes and finally it appears that :
* :heavy_check_mark:  `"mongodb://$mongoHost:$mongoPort/$mongoDB"` works
* :heavy_check_mark: `"mongodb://{$mongoHost}:{$mongoPort}/{$mongoDB}"` works
* :warning:   `"mongodb://${mongoHost}:${mongoPort}/${mongoDB}"` becomes `"mongodb://{mongoHost}:{mongoPort}/{mongoDB}"`.
* :warning:  `"mongodb://{${mongoHost}}:{${mongoPort}}/{${mongoDB}}"` becomes `"mongodb://{${\mongoHost}}:{${\mongoPort}}/{${\mongoDB}}"` .

It's not a very big issue as there are some workaround like using DI string interpolation or stick to the syntax that work, but I guess that could be at least documented. (Or maybe I missed something ?)

**Comment by mnapoli** [comment]
Hey, thanks for the very clear bug report!

I am wondering if this is not an issue inside https://github.com/opis/closure 🤔 The reason I'm saying that is that we use that library to "copy-paste" (in a way) the code inside each closure into the compiled container.

**Comment by n00dl3** [comment]
Ok, I'll try to reproduce with opis/closure alone and let you know.

**Comment by n00dl3** [comment]
Ok, like you can see it comes from opis/closure like you suspected, I saw on open issue about array string interpolation.

I added some test cases. 

Let's see :crossed_fingers: .

Et btw un petit coucou de Villeurbanne :wink: !

**Comment by sorinsarca** [comment]
This was fixed in the latest version of opis/closure `v3.6.2`

**Comment by mnapoli** [comment]
Awesome thank you @sorinsarca and @n00dl3!

> Et btw un petit coucou de Villeurbanne 😉 !

hey hey 👋 ^^
